arangod/IResearch/IResearchLinkHelper.cpp

```diff
@@ -187,13 +189,18 @@ arangodb::Result dropLink( // drop link
   return arangodb::Result();
 }
 
-template<>
-arangodb::Result dropLink<arangodb::iresearch::IResearchViewCoordinator>( // drop link
-    arangodb::LogicalCollection& collection, // link collection
-    arangodb::iresearch::IResearchLink const& link // link to drop
+template <>
+arangodb::Result dropLink<arangodb::iresearch::IResearchViewCoordinator>(  // drop link
+    arangodb::LogicalCollection& collection,        // link collection
+    arangodb::iresearch::IResearchLink const& link  // link to drop
 ) {
-  arangodb::velocypack::Builder builder;
+  // Enterprise variant, we only need to drop links on non-hidden
+  // collections (e.g. in SmartGraph Case)
+  if (arangodb::ClusterMethods::filterHiddenCollections(collection)) {
```

Comment:
how we can handle link properties changes then?

Reply:
Discussed offline, as this is mainly closed Enterprise Code.